### PR TITLE
feat(scala): parse `using` and splatted arguments

### PR DESCRIPTION
**Pull request by brandonspark**
This PR makes it so we can parse `using` and splatted arguments, like `foo(using bar)` and `foo(1, 2, bar*)`.

Splatted arguments was handled in translation, because otherwise it's very hard to tell a priori whether arguments 
end in a PostfixExpr or an Expr.

Parse rate goes to `0.9570153111528416` from `0.95648527803769`

I verified that the splat test case parses to:
```
Pr(
  [DefStmt(
     ({
       name=EN(
              Id(("x", ()),
                {id_info_id=4; id_hidden=false; id_resolved=Ref(None); id_type=Ref(None); id_svalue=Ref(None); }));
       attrs=[KeywordAttr((Const, ()))]; tparams=[]; },
      VarDef(
        {
         vinit=Some(Call(
                      N(
                        Id(("foo", ()),
                          {id_info_id=1; id_hidden=false; id_resolved=Ref(
                           None); id_type=Ref(None); id_svalue=Ref(None); })),
                      [Arg(L(Int((Some(1), ())))); Arg(L(Int((Some(2), ()))));
                       Arg(
                         Call(IdSpecial((Spread, ())),
                           [Arg(
                              N(
                                Id(("bar", ()),
                                  {id_info_id=2; id_hidden=false; id_resolved=Ref(
                                   None); id_type=Ref(None); id_svalue=Ref(
                                   None); })))]))]));
         vtype=None; })))])
```
which is correct.

Closes PA-2688

PR checklist:

- [X] Purpose of the code is [evident to future readers](https://semgrep.dev/docs/contributing/contributing-code/#explaining-code)
- [X] Tests included or PR comment includes a reproducible test plan
- [X] Documentation is up-to-date
- [X] A changelog entry was [added to changelog.d](https://semgrep.dev/docs/contributing/contributing-code/#adding-a-changelog-entry) for any user-facing change
- [X] Change has no security implications (otherwise, ping security team)

If you're unsure about any of this, please see:

- [Contribution guidelines](https://semgrep.dev/docs/contributing/contributing-code)!
- [One of the more specific guides located here](https://semgrep.dev/docs/contributing/contributing/)

### CODE REVIEW

File: languages/scala/recursive_descent/Parser_scala_recursive_descent.ml [hunk 1]
```diff
@@ -2082,28 +2082,39 @@ and multipleArgumentExprs in_ : arguments list =
          | _ -> [])
 
 (** {{{
- *  ArgumentExprs ::= `(` [Exprs] `)`
- *                  | [nl] BlockExpr
+ *  ArgumentExprs     ::= ParArgumentExprs
+ *                      | [nl] BlockExpr
+ *  ParArgumentExprs  ::=  ‘(’ [Exprs] ‘)’
+ *                      |  ‘(’ ‘using’ Exprs ‘)’
+ *                      |  ‘(’ [Exprs ‘,’] PostfixExpr ‘*’ ‘)’ (* TODO *)
```
Comment (emjin) [inline languages/scala/recursive_descent/Parser_scala_recursive_descent.ml:2082]: Did you mean to leave this `TODO` in? If it's indicating that the splatted argument is being handled in scala_to_generic, I think `(* handled in translation *)` would be more clear

File: languages/scala/generic/scala_to_generic.ml [hunk 2]
```diff
@@ -511,9 +522,11 @@ and v_arguments = function
       | Right cases ->
           (lb, [ G.Arg (G.Lambda (cases_to_lambda lb cases) |> G.e) ], rb))
 
-and v_argument v =
+and v_argument ?(is_using = false) v =
```
Comment (emjin) [inline languages/scala/generic/scala_to_generic.ml:525]: If you anticipate changing this, I would just make another function `v_arg_using`. It's not like `v_argument` is very long anyway.

**Review by emjin — APPROVED** [review]
I guess you might want to wait for @aryx but this looks good to me

File: languages/scala/ast/AST_scala.ml [hunk 3]
```diff
@@ -271,6 +271,7 @@ and lhs = expr
 
 and arguments =
   | Args of argument list bracket
+  | ArgUsing of argument list bracket
```
Comment (aryx) [inline languages/scala/ast/AST_scala.ml:274]: I would put ArgUsing of (token (* 'using' *) * expr list) bracket

**Review by aryx — APPROVED** [review]

**Review by aryx — APPROVED** [review]

**Pull request merged**